src/Avalonia.FreeDesktop/DBusSystemDialog.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+using Avalonia.Controls.Platform;
+using Avalonia.Dialogs;
+using Avalonia.Logging;
+using Tmds.DBus;
+
+namespace Avalonia.FreeDesktop
+{
+    internal class DBusSystemDialog : ISystemDialogImpl
+    {
+        private readonly IFileChooser? _fileChooser;
+        private bool _isDbusAvailable;
+
+        internal DBusSystemDialog()
+        {
+            _fileChooser = DBusHelper.Connection?.CreateProxy<IFileChooser>("org.freedesktop.portal.Desktop", "/org/freedesktop/portal/desktop");
+            _isDbusAvailable = _fileChooser is not null;
+        }
+
+        public async Task<string[]?> ShowFileDialogAsync(FileDialog dialog, Window parent)
+        {
+            if (!_isDbusAvailable)
```

Comment:
It would be better to keep DBusSystemDialog only for dbus dialog, and instead add a static factory method "TryCreate" here. And if it fails, then register ManagedSystemDialogImpl. New helper methods won't be needed too.

Reply:
The problem is that Tmds.DBus lazily establishes the dbus connection, so we won't know if the service is available until actually making the call, at which it is too late to register the ManagedSystemDialogImpl, so this redirection is needed.

Reply:
`DBusHelper.TryInitialize`